source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -140,9 +141,27 @@ void AuthenticatorImpl::startVerify() {
   curr_token_ = std::move(tokens_.back());
   tokens_.pop_back();
 
-  jwt_ = std::make_unique<::google::jwt_verify::Jwt>();
-  ENVOY_LOG(debug, "{}: Parse Jwt {}", name(), curr_token_->token());
-  Status status = jwt_->parseFromString(curr_token_->token());
+  if (provider_) {
+    jwks_data_ = jwks_cache_.findByProvider(provider_.value());
+    if (jwks_data_->getJwtProvider().enable_token_cache()) {
+      jwt_ = jwks_data_->getTokenCache().find(curr_token_->token());
+      if (jwt_) {
+        use_cache_jwt_ = true;
+        jwks_data_->getTokenCache().release(curr_token_->token(), jwt_);
```

Comment:
No,  if you release it now,  the cached item may be freed by other operations due to cache is full  while this auth is paused and waiting for fetching jwks.    It is a data corruption, jwt_ is pointing to a wrong data.

Reply:
done.